src/Core/Checkout/Order/Event/OrderCriteriaEvent.php

```diff
@@ -0,0 +1,26 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Order\Event;
+
+use Shopware\Core\Framework\DataAbstractionLayer\Search\Criteria;
+use Shopware\Core\System\SalesChannel\SalesChannelContext;
+use Symfony\Contracts\EventDispatcher\Event;
+
+class OrderCriteriaEvent extends Event
+{
+    public function __construct(
+        private readonly Criteria $criteria,
+        private readonly SalesChannelContext $context
+    ) {
+    }
+
+    public function getCriteria(): Criteria
+    {
+        return $this->criteria;
+    }
+
+    public function getContext(): SalesChannelContext
```

Comment:
This would be better, wouldn't it? Like in many other Events
```php
public function getContext(): Context
{
    return $this->salesChannelContext->getContext();
}

public function getSalesChannelContext(): SalesChannelContext
{
    return $this->salesChannelContext;
}
```

Reply:
I just adjusted the code.